linux/runtime.go

```diff
@@ -160,23 +247,40 @@ func (r *Runtime) Delete(ctx context.Context, c plugin.Task) (*plugin.Exit, erro
 		return nil, err
 	}
 	lc.shim.Exit(ctx, &shim.ExitRequest{})
+	r.tasks.delete(ctx, lc)
 	return &plugin.Exit{
 		Status:    rsp.ExitStatus,
 		Timestamp: rsp.ExitedAt,
 	}, r.deleteBundle(namespace, lc.containerID)
 }
 
 func (r *Runtime) Tasks(ctx context.Context) ([]plugin.Task, error) {
-	dir, err := ioutil.ReadDir(r.root)
+	namespace, err := namespaces.NamespaceRequired(ctx)
```

Comment:
@mlaventure yes. That sort of action should be restricted to administrative tools since most tools responsible for running containers would only operate in a single namespace. Additionally namespaces should be able to be used to limit access to resources without pushing that duty on the client to filter results.